choreolib/src/main/java/com/choreo/lib/ChoreoAutoTrajectory.java

```diff
@@ -0,0 +1,428 @@
+  }
+
+  /**
+   * Creates a command that allocates the drive subsystem and follows the trajectory using the
+   * factories control function
+   *
+   * @return The command that will follow the trajectory
+   */
+  public Command cmd() {
+    return new FunctionalCommand(
+        this::cmdInitialize, this::cmdExecute, this::cmdEnd, this::cmdIsFinished, driveSubsystem);
+  }
+
+  /**
+   * Will get the starting pose of the trajectory.
+   *
+   * <p>This position is mirrored based on the {@code mirrorTrajectory} boolean supplier in the
+   * factory used to make this trajectory
+   *
+   * @return The starting pose
+   */
+  public Pose2d getStartingPose() {
+    if (mirrorTrajectory.getAsBoolean()) {
+      return trajectories.get(0).getFlippedInitialPose();
+    } else {
+      return trajectories.get(0).getInitialPose();
+    }
+  }
+
+  /**
+   * Returns a trigger that is true while the command is scheduled.
+   *
+   * @return A trigger that is true while the command is scheduled.
+   */
+  public Trigger active() {
+    return new Trigger(loop, () -> this.isActive);
+  }
+
+  /**
+   * Returns a trigger that is true while the command is not scheduled.
+   *
+   * <p>The same as calling <code>active().negate()</code>.
+   *
+   * @return A trigger that is true while the command is not scheduled.
+   */
+  public Trigger inactive() {
+    return active().negate();
+  }
+
+  /**
+   * Returns a trigger that is true when the command is finished.
+   *
+   * @return A trigger that is true when the command is finished.
+   */
+  public Trigger done() {
+    return new Trigger(loop, () -> this.isDone);
+  }
+
+  /**
+   * Returns a trigger that will go true for 1 cycle when the desired time has elapsed
+   *
+   * @param timeSinceStart The time since the command started in seconds.
+   * @return A trigger that is true when timeSinceStart has elapsed.
+   */
+  public Trigger atTime(double timeSinceStart) {
+
+    // The timer shhould never be negative so report this as a warning
+    if (timeSinceStart < 0) {
+      DriverStation.reportWarning("Trigger time cannot be negative for " + name, true);
+      return offTrigger;
+    }
+
+    // The timer should never exceed the total trajectory time so report this as a warning
+    if (timeSinceStart > totalTime()) {
+      DriverStation.reportWarning(
+          "Trigger time cannot be greater than total trajectory time for " + name, true);
+      return offTrigger;
+    }
+
+    // Make the trigger only be high for 1 cycle whne the time has elapsed,
+    // this is needed for better support of multi-time triggers for multi events
+    return new Trigger(
+        loop,
+        new BooleanSupplier() {
+          double lastTimestamp = timer.get();
+
+          public boolean getAsBoolean() {
+            double nowTimestamp = timer.get();
+            return lastTimestamp < nowTimestamp && nowTimestamp >= timeSinceStart;
+          }
+        });
```

Comment:
I don't think this works as intended. `lastTimestamp < nowTimestamp` will always be true because getAsBoolean always happens after the creation of the BooleanSupplier.